pkg/transportrequest/uploadcts.go

```diff
@@ -0,0 +1,233 @@
+func getFioriDeployStatement(
+	transportRequestID string,
+	configFile string,
+	app CTSApplication,
+	cts CTSConnection,
+) (string, error) {
+	desc := app.Desc
+	if len(desc) == 0 {
+		desc = "Deployed with Piper based on SAP Fiori tools"
+	}
+
+	useConfigFile, noConfig, err := handleConfigFile(configFile)
+	if err != nil {
+		return "", err
+	}
+	cmd := []string{
+		"fiori",
+		"deploy",
+		"--failfast", // provide return code != 0 in case of any failure
+		"--yes",      // autoconfirm --> no need to press 'y' key in order to confirm the params and trigger the deployment
+		"--username", abapUserKey,
+		"--password", abapPasswordKey,
+		"--description", fmt.Sprintf("\"%s\"", desc),
+	}
+
+	if noConfig {
+		cmd = append(cmd, "--noConfig") // no config file, but we will provide our parameters
+	}
+	if useConfigFile {
+		cmd = append(cmd, "--config", fmt.Sprintf("\"%s\"", configFile))
+	}
+	if len(cts.Endpoint) > 0 {
+		log.Entry().Debugf("Endpoint '%s' used from piper config", cts.Endpoint)
+		cmd = append(cmd, "--url", cts.Endpoint)
+	} else {
+		log.Entry().Debug("No endpoint found in piper config.")
+	}
+	if len(cts.Client) > 0 {
+		log.Entry().Debugf("Client '%s' used from piper config", cts.Client)
+		cmd = append(cmd, "--client", cts.Client)
+	} else {
+		log.Entry().Debug("No client found in piper config.")
+	}
+	if len(transportRequestID) > 0 {
+		log.Entry().Debugf("TransportRequestID '%s' used from piper config", transportRequestID)
+		cmd = append(cmd, "--transport", transportRequestID)
+	} else {
+		log.Entry().Debug("No transportRequestID found in piper config.")
+	}
+	if len(app.Pack) > 0 {
+		log.Entry().Debugf("application package '%s' used from piper config", app.Pack)
+		cmd = append(cmd, "--package", app.Pack)
+	} else {
+		log.Entry().Debug("No application package found in piper config.")
+	}
+	if len(app.Name) > 0 {
+		log.Entry().Debugf("application name '%s' used from piper config", app.Name)
+		cmd = append(cmd, "--name", app.Name)
+	} else {
+		log.Entry().Debug("No application name found in piper config.")
+	}
+
+	return strings.Join(cmd, " "), nil
+}
+
+func getSwitchUserStatement(user string) string {
+	return fmt.Sprintf("su %s", user)
+}
+
+func handleConfigFile(path string) (useConfigFile, noConfig bool, err error) {
+
+	if len(path) == 0 {
+		exists, e := files.FileExists(defaultConfigFileName)
+		if e != nil {
+			return false, false, e
+		}
+		useConfigFile = false
+		noConfig = !exists
+		return
+	}
+	exists, err := files.FileExists(path)
+	if err != nil {
+		return false, false, err
+	}
+	if exists {
+		useConfigFile = true
+		noConfig = false
+	} else {
+		if path != defaultConfigFileName {
+			return false, false, fmt.Errorf("Configured deploy config file '%s' does not exists", path)
+		}
+		// in this case this is most likely provided by the piper default config and
+		// it was not explicitly configured. Hence we assume not having a config file
+		useConfigFile = false
+		noConfig = true
+	}
+	return
+}
```

Comment:
So... I'm getting through your unit tests with this:
```suggestion
func handleConfigFile(path string) (useConfigFile, noConfig bool, err error) {
	useConfigFile, err = configFileExists(path)
	if err == nil {
		noConfig, err =	noConfigFile(path)
	}

	return
}

func configFileExists(path string) (exists bool, err error) {
	if len(path) != 0 {
		exists, err = files.FileExists(path)
	}
	return
}

func noConfigFile(path string) (noConfig bool, err error) {
	exists := false

	if len(path) == 0 {
		exists, err = files.FileExists(defaultConfigFileName)
	} else {
		exists, err = files.FileExists(path)
		if exists == false && path != defaultConfigFileName {
			err = fmt.Errorf("Configured deploy config file '%s' does not exists", path)
			return
		}
	}
	noConfig = !exists

	return
}
```

Reply:
More understandable name for the func introduced, also for the variables. Regarding splitting it up in several functions, there are - as always in life pros and cons. Not strongly against that. The current approach allows to see how the two params relates to each other ...

Reply:
Regarding the split into three funcs: With that we would have one method for returning the two booleans. Internally we use the other two funcs for getting the value of the boolean. If we go the way with the individual funcs for calculating the boolean we could also use the two funcs directly without having the third method returning the two booleans in-between. Having that two-boolean func is not end it itself.

My personal opinion is: both (having individual methods for returning each one boolean or having one func returning both boolean) is technically possible and not that far from each other. I personally are in favour of having one method returning two booleans (the booleans related  to each other, we don't repeat a len-check), but the other approach is also valid. I think when having two valid approaches it does not make sense to start a fight in the details. We have more important things to do.